src/test/java/com/nvidia/spark/rapids/jni/iceberg/IcebergDateTimeUtilTest.java

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.jni.iceberg;
+
+import static ai.rapids.cudf.AssertUtils.assertColumnsAreEqual;
+
+import java.time.LocalDate;
+import java.util.Random;
+
+import org.junit.jupiter.api.Test;
+import org.apache.iceberg.util.DateTimeUtil;
+import ai.rapids.cudf.ColumnVector;
+
+public class IcebergDateTimeUtilTest {
+
+  @Test
+  void toYearsTest() {
+    // basic test
+    try (
+        ColumnVector input = ColumnVector.timestampDaysFromBoxedInts(
+            -50,
+            -500,
+            null,
+            50,
+            500,
+            null);
+        ColumnVector expected = ColumnVector.fromBoxedInts(
+            -1,
+            -2,
+            null,
+            0,
+            1,
+            null);
+        ColumnVector result = IcebergDateTimeUtil.toYears(input)) {
+      assertColumnsAreEqual(expected, result);
+    }
+
+    Random random = new Random();
+    int numRows = 1024;
```

Comment:
We should considering making them a static variable, init in a `beforeAll` method, and logging the seed so that when an error happens, we should reproduce it locally.